test/model.create.test.js

```diff
@@ -197,6 +197,27 @@ describe('model', function() {
         const docs = await Test.find();
         assert.equal(docs.length, 5);
       });
+      it('should throw an error only after all the documents have finished saving gh-4628', async function() {
+        const countSchema = new Schema({ n: Number });
+        const testSchema = new Schema({ name: { type: String, unique: true }, reference: Number });
+
+        const Count = db.model('gh4628', countSchema);
+
+        testSchema.pre('save', async function(next) {
```

Comment:
Why is this pre hook here? Doesn't seem to be necessary. Are you just looking for a way to leave some saves in progress to test?

Reply:
I just copied the above test and modified it slightly. I see now that for this test that part is not necessary.